src/components/options/providers/main.svelte

```diff
@@ -0,0 +1,44 @@
+<script>
+import AddSearchProviders from "./add.svelte";
+import ContextMenuItems from "./contextMenuItems.svelte";
+import Groups from "./groups.svelte";
+import { onMount } from "svelte";
+
+let groupComponent;
+let contextMenuComponent;
+
+function mainConfigurationUpdated() {
+  updateForms();
+  chrome.runtime.sendMessage({ action: "updateContextualMenu" });
+}
+
+function updateForms() {
+  return Promise.all([groupComponent.initialize(), updateProvidersForm()]);
+}
+
+async function updateProvidersForm() {
+  contextMenuComponent.initProvidersAndGroups();
+}
+
+onMount(() => {
+  updateForms();
+});
+</script>
+
+<div>
+  <h2>Add Search Provider</h2>
+  <AddSearchProviders on:updateMainConfiguration="{mainConfigurationUpdated}" />
+
+  <h2 class="mt-4">Manage Context Menu Items</h2>
+  <ContextMenuItems
+    bind:this="{contextMenuComponent}"
+    on:updateMainConfiguration="{mainConfigurationUpdated}" />
+
+  <h2 class="mt-3">Manage Groups</h2>
+  <Groups
+    bind:this="{groupComponent}"
+    {...$$props}
+    on:updateMainConfiguration="{mainConfigurationUpdated}"
+    on:updateMainConfiguration="{updateForms}"
+    on:updateMainConfiguration="{updateProvidersForm}" />
```

Comment:
This code is very confusing to me: you are adding _three_ listeners to the `updateMainConfiguration` event to execute `mainConfigurationUpdated`, `updateForms` and `updateProvidersForm` **but** `mainConfigurationUpdated` already calls `updateForms` (within is code) and `updateForms` already calls `updateProvidersForms` (within his code).

Are you sure that the only listener here shouldn't be `on:updateMainConfiguration="{mainConfigurationUpdated}"`?

Reply:
Yes you are right, after changing the function I forgot to remove the listeners.

Reply:
Did you committed your changes? The three listeners seems to still be there...